tests/modules/test_k8s_prerequisites_check.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+from unittest.mock import Mock
+
+from rookify.modules.exception import ModuleException
+from typing import Any, Dict, List
+from ..mock_k8s_prerequisite_check import MockK8sPrerequisitesCheckHandler
+
+
+# Note: currently this test works with pytest but not with unittest, which is not able to import needed classes
+class TestK8sPrerequisitesCheckHandler(unittest.TestCase):
+    def setUp(self) -> None:
+        # Mock configuration
+        self.config = {"rook": {"cluster": {"namespace": "test-namespace"}}}
+        # No response option
+        self.empty_response = False
+
+    def _request_callback(
+        self, method: str, *args: List[Any], **kwargs: Dict[Any, Any]
+    ) -> Any:
+        if method == "apps_v1_api.list_deployment_for_all_namespaces":
+            if self.empty_response is True:
+                return MockResponse([])
+            return MockResponse(["apple", "banana", "cherry"])
+        if method == "core_v1_api.list_namespace":
+            return self._mock_list_namespace()
+
+    def _mock_list_namespace(self) -> Any:
```

Comment:
Same here. You can reuse Python `kubernetes-client` objects for that purpose :)

Reply:
ok thanks! good idea, used kubernets-client namespace options to create the mock now